controller/scheduler/fleet.py

```diff
@@ -271,14 +271,14 @@ def run(self, name, image, entrypoint, command):  # noqa
             tran = ssh.get_transport()
 
             def _do_ssh(cmd):
-                chan = tran.open_session()
-                # get a pty so stdout/stderr look right
-                chan.get_pty()
-                out = chan.makefile()
-                chan.exec_command(cmd)
-                output = out.read()
-                rc = chan.recv_exit_status()
-                return rc, output
+                with tran.open_session() as chan:
+                    chan.exec_command(cmd)
+                    while not chan.exit_status_ready():
+                        time.sleep(1)
```

Comment:
Yes, that's probably better than my infinite looping (which will be killed after the gunicorn task timeout eventually I assume: 20 min.).